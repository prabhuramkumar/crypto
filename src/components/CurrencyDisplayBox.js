import React from 'react';
import {convertToAMPM, formatDate} from '../utils/displayUtils';

const CurrencyDisplayBox = (props)=>{
    return(
        <div id={props.key} className="currencyDisplay__box">
            <div className="currencyDisplay__box--date">{formatDate(props.currency.date)}</div>
            <div className="currencyDisplay__box--name">{props.currency.currency}</div>
            <div className="currencyDisplay__box--section">
                <div>
                    <div className="currencyDisplay__box--transaction">Buy</div>
                    <div className="currencyDisplay__box--price">${props.currency.buyValue}</div>
                    <div className="currencyDisplay__box--time">{convertToAMPM(props.currency.buyTime)}</div>
                </div>
                <div>
                    <div className="currencyDisplay__box--transaction">Sell</div>
                    <div className="currencyDisplay__box--price">${props.currency.sellValue}</div>
                    <div className="currencyDisplay__box--time">{convertToAMPM(props.currency.sellTime)}</div>
                </div>
            </div>
            <div className="currencyDisplay__box--profit">${props.currency.profit}</div>
        </div>
    )
}

export default CurrencyDisplayBox;