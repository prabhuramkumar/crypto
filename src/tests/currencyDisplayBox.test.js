import React from 'react';
import { shallow, mount } from 'enzyme';
import CurrencyDisplayBox from '../components/CurrencyDisplayBox';
describe('currencyDisplayBox', () => {
  it('should render correctly in normal mode', () => {
  	const currency= {
      "currency": "BTC",
      "buyValue": "34.98",
      "sellValue": "37.01",
      "buyTime":  "0915",
      "sellTime": "1230",
      "date": "20180507",
      "profit": "2.03"
    }
    const currencyDisplayBox = shallow(<CurrencyDisplayBox currency={currency}  key={currency.id} />);
  	expect(currencyDisplayBox).toMatchSnapshot();
  });
});

describe('currencyDisplayBox', () => {
  it('should render correctly in "debug" mode', () => {
  	const currency= {
      "currency": "BTC",
      "buyValue": "34.98",
      "sellValue": "37.01",
      "buyTime":  "0915",
      "sellTime": "1230",
      "date": "20180507",
      "profit": "2.03"
    }
    const currencyDisplayBox = shallow(<CurrencyDisplayBox currency={currency}  key={currency.id} />);
  	const component = currencyDisplayBox.find('.currencyDisplay__box');
  	const price = currencyDisplayBox.find('.currencyDisplay__box--price');
  	const time = currencyDisplayBox.find('.currencyDisplay__box--time');
  	const profit = currencyDisplayBox.find('.currencyDisplay__box--profit');
    expect(component.length).toEqual(1);
    expect(price[0].text()).toEqual('$34.98');
    expect(price[1].text()).toEqual('$37.01');
    expect(time[0].text()).toEqual('09:15 AM');
    expect(time[1].text()).toEqual('12:30 PM');
    expect(profit.text()).toEqual('$2.03');
  });
});

