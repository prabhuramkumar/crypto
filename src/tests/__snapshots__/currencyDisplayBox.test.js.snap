// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencyDisplayBox should render correctly in normal mode 1`] = `
<div
  className="currencyDisplay__box"
>
  <div
    className="currencyDisplay__box--date"
  >
    2018-05-07
  </div>
  <div
    className="currencyDisplay__box--name"
  >
    BTC
  </div>
  <div
    className="currencyDisplay__box--section"
  >
    <div>
      <div
        className="currencyDisplay__box--transaction"
      >
        Buy
      </div>
      <div
        className="currencyDisplay__box--price"
      >
        $
        34.98
      </div>
      <div
        className="currencyDisplay__box--time"
      >
        09:15 AM
      </div>
    </div>
    <div>
      <div
        className="currencyDisplay__box--transaction"
      >
        Sell
      </div>
      <div
        className="currencyDisplay__box--price"
      >
        $
        37.01
      </div>
      <div
        className="currencyDisplay__box--time"
      >
        12:30 PM
      </div>
    </div>
  </div>
  <div
    className="currencyDisplay__box--profit"
  >
    Profit: $
    2.03
  </div>
</div>
`;
